### PR TITLE
feat(autocomplete): add value support

### PR DESCRIPTION
**Pull request by kara**
Autocomplete is WIP. This is a small piece to add value population.

Note: The placeholder animation will still happen when setting the value. This will be fixed in a later PR once floating placeholders are more configurable in input.

### CODE REVIEW

File: src/lib/autocomplete/autocomplete-trigger.ts [hunk 1]
```diff
@@ -84,6 +98,23 @@ export class MdAutocompleteTrigger implements OnDestroy {
...
   }
 
+   /**
+   * This method closes the panel, and if a value is specified, also sets the associated
+   * control to that value. It will also mark the control as dirty if this interaction
+   * stemmed from the user.
+   */
+  private _setValueAndClose(event: MdOptionSelectEvent | null): void {
+    if (event) {
+      // TODO(kara): revisit animation once floating placeholder is toggle-able
```
Comment (jelbourn) [inline src/lib/autocomplete/autocomplete-trigger.ts]: `TODO` probably isn't necessary any more

File: src/lib/autocomplete/autocomplete.spec.ts [hunk 2]
```diff
@@ -135,20 +164,78 @@ describe('MdAutocomplete', () => {
...
+      expect(fixture.componentInstance.stateCtrl.dirty)
+          .toBe(false, `Expected control to start out pristine.`);
+
+      fixture.componentInstance.stateCtrl.setValue('AL');
+      fixture.detectChanges();
+
+      expect(fixture.componentInstance.stateCtrl.dirty)
+          .toBe(false, `Expected control to stay pristine if value is set programmatically.`);
+    });
+
```
Comment (jelbourn) [inline src/lib/autocomplete/autocomplete.spec.ts]: Should there be any tests for touched state?

File: src/demo-app/autocomplete/autocomplete-demo.html [hunk 3]
```diff
@@ -1,9 +1,52 @@
...
+      <button md-button (click)="stateCtrl.setValue('California')">SET VALUE</button>
+      <button md-button (click)="stateCtrl.enabled ? stateCtrl.disable() : stateCtrl.enable()">
+        TOGGLE DISABLED
+      </button>
+    </md-card-actions>
+
+  </md-card>
+
+  <md-card>
+    <div>TD value (currentState): {{ currentState }}</div>
```
Comment (kara) [inline src/demo-app/autocomplete/autocomplete-demo.html]: Fixed to "template-driven" for clarity

File: src/lib/autocomplete/autocomplete.spec.ts (hunk 2 above)
Comment (kara) [inline src/lib/autocomplete/autocomplete.spec.ts]: That should be tested by the forms module and/or mdInput. The autocomplete doesn't actually touch those events, so seems duplicative to test here.

**Comment by kara** [comment]
@jelbourn Comments addressed!

**Pull request merged**